src/pages/FollowUpAnalysis/FollowUpAnalysis.js

```diff
@@ -0,0 +1,218 @@
+import { useSidebar } from '../../components/Sidebar/SidebarContext'
+import { convertDateToApiDateFormat } from '../../helpers/dates'
+import threeMonthsAgo from '../../helpers/threeMonthsAgo'
+import { apiConf } from '../../server.conf'
+import cssPageStyles from '../Page.module.css'
+import FollowUpAnalysisTable from './FollowUpAnalysisTable/FollowUpAnalysisTable'
+import Form from './Form'
+
+const generateElementKey = e =>
+    `${e.attributeOptionComboId}-${e.categoryOptionComboId}-${e.periodId}-${e.sourceId}-${e.dataElementId}`
+
+const convertElementFromApiResponse = e => ({
+    key: generateElementKey(e),
+    attributeOptionComboId: e.attributeOptionComboId,
+    categoryOptionComboId: e.categoryOptionComboId,
+    periodId: e.periodId,
+    organisationUnitId: e.sourceId,
+    dataElementId: e.dataElementId,
+    dataElement: e.dataElementName,
+    organisation: e.sourceName,
+    period: e.period.name,
+    min: e.min,
+    max: e.max,
+    value: Number.parseFloat(e.value, 10),
+    marked: !e.followup,
+    comment: e.comment,
+})
+
+const convertElementToUnFollowupRequest = e => ({
+    dataElementId: e.dataElementId,
+    periodId: e.periodId,
+    organisationUnitId: e.organisationUnitId,
+    categoryOptionComboId: e.categoryOptionComboId,
+    attributeOptionComboId: e.attributeOptionComboId,
+    followup: false,
+})
+
+const useFormState = () => {
+    const [startDate, setStartDate] = useState(threeMonthsAgo())
+    const [endDate, setEndDate] = useState(new Date())
+    const [organisationUnitId, setOrganisationUnitId] = useState(null)
+    const [dataSetIds, setDataSetIds] = useState([])
+
+    const handleStartDateChange = (event, date) => {
+        setStartDate(new Date(date))
+    }
+    const handleEndDateChange = (event, date) => {
+        setEndDate(new Date(date))
+    }
+    const handleDataSetsChange = ({ selected }) => {
+        setDataSetIds(selected)
+    }
+
+    return {
+        startDate,
+        handleStartDateChange,
+        endDate,
+        handleEndDateChange,
+        organisationUnitId,
+        handleOrganisationUnitChange: setOrganisationUnitId,
+        dataSetIds,
+        handleDataSetsChange,
+    }
+}
+
+const FollowUpAnalysis = ({ sectionKey }) => {
+    const sidebar = useSidebar()
+    const [loading, setLoading] = useState(false)
+    const [tableVisible, setTableVisible] = useState(false)
+    const [elements, setElements] = useState([])
+    const { d2 } = useD2()
+    const {
+        startDate,
+        handleStartDateChange,
+        endDate,
+        handleEndDateChange,
+        organisationUnitId,
+        handleOrganisationUnitChange,
+        dataSetIds,
+        handleDataSetsChange,
+    } = useFormState()
+    const noValuesFoundAlert = useAlert(i18n.t('No values found'), {
+        success: true,
+    })
+    const successfulUnfollowAlert = useAlert(i18n.t('Elements unfollowed'), {
+        success: true,
+        duration: 2e3,
```

Comment:
I think I'd just go for 2000 here. `2e3` is more concise, but when scanning through the code I'd say that 2000 is easier to parse.

When you're dealing with say, a million or larger, where you're actually not able to see the amount of zeros at a glance, I can see the advantage. But for smaller numbers just writing it out seems more readable to me.